tesp/sensitivity.py

```diff
@@ -32,33 +37,410 @@ class ExperimentList(luigi.WrapperTask):
+
+
+def sum_and_count(src_product, band_name):
+    """
+    Sum and count of pixels of surface reflectance images.
+    """
+    src_ds = find_dataset_by_name(src_product, band_name)
+    src_count = find_dataset_by_name(src_product, band_name + "_pixelcount")
+
+    src_data = src_ds[:]
+    if src_count is None:
+        count_data = numpy.where(src_data == src_ds.attrs['no_data_value'], 0, 1)
+        src_valid_data = numpy.where(count_data > 0, src_data, 0)
+        src_data = src_valid_data
+    else:
+        count_data = src_count[:]
+
+    return src_data, count_data
+
+
+def copy_dataset(src_product, target_product, band_name):
+    """
+    If only one dataset has this band, then just copy it over to the target dataset.
+    """
+    src_ds = find_dataset_by_name(src_product, band_name)
+
+    src_data, count_data = sum_and_count(src_product, band_name)
+
+    target_ds = target_product.create_dataset(band_name, data=src_data, dtype='int32', chunks=(1024, 1024))
+    target_count = target_product.create_dataset(band_name + "_pixelcount",
+                                                 data=count_data, dtype='int32', chunks=(1024, 1024))
+
+    for key in src_ds.attrs:
+        target_ds.attrs[key] = src_ds.attrs[key]
+        target_count.attrs[key] = src_ds.attrs[key]
+
+    target_count.attrs['alias'] = band_name + "_pixelcount"
+
+
+def merge_datasets(left_product, right_product, target_product, band_name):
+    """
+    Merge images from two `.h5` groups in two different `.h5` container.
+    """
+    left_ds = find_dataset_by_name(left_product, band_name)
+    right_ds = find_dataset_by_name(right_product, band_name)
+
+    assert left_ds.attrs['crs_wkt'] == right_ds.attrs['crs_wkt'], "I can't merge images from different CRSs yet"
+    left_affine = Affine.from_gdal(*left_ds.attrs['geotransform'])
+    right_affine = Affine.from_gdal(*right_ds.attrs['geotransform'])
+    assert numpy.allclose(linear_part(left_affine), linear_part(right_affine))
+
+    target_affine = combine_affine(left_affine, right_affine)
+
+    left_end = ~target_affine * (left_affine * left_ds.shape)
+    right_end = ~target_affine * (right_affine * right_ds.shape)
+    target_shape = (int(max(left_end[0], right_end[0])), int(max(left_end[1], right_end[1])))
+
+    target_ds = target_product.create_dataset(band_name, shape=target_shape, dtype='int32', chunks=(1024, 1024))
+    target_ds[:] = 0
+    target_count = target_product.create_dataset(band_name + "_pixelcount",
+                                                 shape=target_shape, dtype='int32', chunks=(1024, 1024))
+    target_count[:] = 0
+
+    def add_image(src_affine, src_product):
+        """
+        Add this image to the target array.
+        """
+        src_ds = find_dataset_by_name(src_product, band_name)
+
+        # find the target window
+        ul = ~target_affine * (src_affine * (0, 0))
+        lr = ~target_affine * (src_affine * src_ds.shape)
```

Comment:
Careful with this line
src_affine * src_ds.shape  looks incorrect.  src_ds.shape will return (y, x) , but requires affine * (x, y)

If your shape is square then ok, but probably safer to not be specific

Reply:
Thanks! This was it! Seems to work now, asking for your review again.

This time there is a class called `GeoBox` which may someday get merged back into `GriddedGeoBox`.